domainbed/algorithms.py

```diff
@@ -1007,22 +1007,23 @@ def update(self, minibatches, unlabeled=False):
             env_loss = F.cross_entropy(logits, y)
             total_loss += env_loss
 
-            grads.append( autograd.grad(env_loss, self.network.parameters(), retain_graph=True) )
+            grads.append( autograd.grad(env_loss, self.network.parameters(), retain_graph=True, create_graph=True) )
             
         mean_loss = total_loss / len(minibatches)
-        mean_grad = autograd.grad(mean_loss, self.network.parameters(), retain_graph=True)
+        mean_grad = autograd.grad(mean_loss, self.network.parameters(), retain_graph=True, create_graph=True)
 
         # compute trace penalty
         penalty_value = 0
         for grad in grads:
             for g, mean_g in zip(grad, mean_grad):
```

Comment:
I think there is a bug here. Since `mean_grad` is only one element, your `zip` will end after one iteration. Example:

```python
l1 = [1, 2, 3]
l2 = [4]

for a, b in zip(l1, l2):
    print(a, b)
```

Prints `1 4`.